commonTools/build_stats/BuildStatsWrappers.cmake

```diff
@@ -74,34 +181,44 @@ function(generate_build_stats_wrapper_for_op op_name variable_to_set)
 
   # there's an issue here - if CMAKE_FOO is unset (whatever `variable_to_set` is)
   # we need a to know the command - but CMake hasn't chosen one yet...
-  if ("${${variable_to_set}}" STREQUAL "")
+  if( (ALLOW_FIND) AND ( ("${${variable_to_set}}" STREQUAL "")
+                         OR
+                         (NOT ${variable_to_set}) ))
     message("-- " "${variable_to_set} is not set, but a wrapper has been requested. Asking CMake to find ${op_lc}")
     find_program(${variable_to_set} "${op_lc}")
     print_var(${variable_to_set})
   endif()
 
-  # Override the compiler with the wrapper but remember the original compiler
-  # we take as a paramter a CMake variable to set, e.g., CMAKE_CXX_COMPILER
-  if ("${${variable_to_set}_ORIG}" STREQUAL "")
-    # we want to set CMAKE_C_COMPILER_ORIG to CMAKE_C_COMPILER
-    # variable to set is a string, so we need the value of the string evaluated
+  # Override the op with the wrapper but remember the original command
+  # we take as a parameter a CMake `variable_to_set`, e.g., CMAKE_CXX_COMPILER
+  # we only wrap if `variable_to_set` is defined.
+  if( ("${${variable_to_set}_ORIG}" STREQUAL "") AND (${variable_to_set}) )
```

Comment:
I haven't messed with reconfigure testing at all.

I'm guessing the logic is convoluted at this point. As pointed out elsewhere, the checks for TIME_CMD are redundant, since I also apply that check at the top-most level.

I realistically won't be able to address any of these until Thursday or Friday (so 4-5 days out).

Reply:
This is fixed in my branch in the PR I will post shortly.